cmd/crank/beta/validate/manager.go

```diff
@@ -171,21 +184,30 @@ func (m *Manager) CacheAndLoad(cleanCache bool) error {
 
 func (m *Manager) addDependencies() error {
 	for image := range m.confs {
-		m.deps[image] = true // we need to download the configuration package for the XRDs
+		cfg := &metav1.Configuration{}
 
-		layer, err := m.fetcher.FetchBaseLayer(image)
-		if err != nil {
-			return errors.Wrapf(err, "cannot download package %s", image)
-		}
+		switch c := m.confs[image].(type) {
+		case bool:
```

Comment:
from this switch statement, i can tell that we expect `bool` or `*metav1.Configuration` to be included in this map. interestingly enough, both of these types will end up with a usable `*metav1.Configuration` after processing occurs in this switch.

Would it make more sense to model `m.confs` as `map[string]*metav1.Configuration` instead? when we find just the image we could insert the string key with a nil value. When we find the image and a `*metav1.Configuration` we can insert with a non-nil value.

that way, we can tell a couple things while using a more strict limit on the types we allow/expect that is arguably easier to read and reason about.

1. we know we have seen an image before if the key is present 
2. we know we have a final `*metav1.Configuration` to use if the value is nil or not

would that work better in your opinion? 🤔

Reply:
This seems easier to read and understand but with that approach we will be assigning each of the configuration we downloaded to the `m.confs` map. 
I initially designed my approach as you described but it felt like we might use too much memory in cases where user loads too much configuration but looking at it now I also feel like a bunch of configuration yaml files wouldn't be a problem for us in the memory department if we dont have like 1000 of them. I will change this approach to your proposed one.